troubadix/plugins/script_version_and_last_modification_tags.py

```diff
@@ -129,6 +129,41 @@ def check_content(
                 file=nasl_file,
                 plugin=self.name,
             )
+        else:
+            date_str = match_last_modified.group("value")
+            format_left = "%Y-%m-%d %H:%M:%S %z "
+            format_right = "(%a, %d %b %Y)"
+            try:
+                date_left = datetime.datetime.strptime(
+                    date_str[:26], format_left
+                )
+                date_right = datetime.datetime.strptime(
+                    date_str[26:], format_right
+                )
+                week_day_parsed = date_right.strftime("%a")
+            except ValueError:
+                yield LinterError(
+                    "False or incorrectly formatted modification_date.",
+                    file=nasl_file,
+                    plugin=self.name,
+                )
+                return
+            week_day_str = date_str[27:30]
+            # Wed, 29 Nov 2017
+            if date_left.date() != date_right.date():
```

Comment:
Why can't we compare the two `datetime` objects directly?

Reply:
Because the date object contains the same date twice, and the format then contains capture targets like the year %y twice. And this library only works with parsing a single target from a date object once using capture groups. It then complains that capture groups are being redefined. [related pull request explaining why it doesn't work](https://github.com/python/cpython/issues/48680)

Reply:
I actually meant why we can't just `date_left == date_right`, but missed that this obviously cannot work, because the left contains hours, minutes, ..., while the right doesn't. Nevermind then :)